examples/Processor.hs

```diff
@@ -77,29 +94,40 @@ data RW a = R Key                 (Value -> a)
 -- | A program is a free selective on top of the 'RW' base functor.
 type Program a = Select RW a
 
-instance Show a => Show (RW a) where
+instance Show (RW a) where
     show (R k          _) = "Read "  ++ show k
     show (W k (Pure v) _) = "Write " ++ show k ++ " " ++ show v
     show (W k _        _) = "Write " ++ show k
 
+logEntry :: MonadState State m => LogEntry Key Value -> m ()
+logEntry item = S.modify $ \s ->
```

Comment:
Maybe this is where the confusion about the order of log entries comes from -- you prepend to the log, rather than appending to it?

Reply:
Indeed, that's the reason. Thanks!